### PR TITLE
que3.2

### PR DESCRIPTION
**Pull request by ashok-choudhary-0**

### CODE REVIEW

File: controllers/userController.js [hunk 1]
```diff
@@ -129,5 +129,24 @@ const userAddress = async (req, res) => {
     res.status(500).send({ message: err })
   }
 }
-
-module.exports = { userRegister, login, getUserData, deleteUserData, limitUsersData, userAddress }
+const deleteUserAddress = async (req, res) => {
+  const { addressArray, user_id } = req.body
```
Comment (amrit56789) [inline controllers/userController.js:133]: take a only user_id in array from postman

File: controllers/userController.js [hunk 2]
```diff
@@ -129,5 +129,24 @@ const userAddress = async (req, res) => {
...
+  try {
+    if (addressArray.length == 0) {
```
Comment (amrit56789) [inline controllers/userController.js:135]: remove this

File: controllers/userController.js [hunk 3]
```diff
@@ -129,5 +129,24 @@ const userAddress = async (req, res) => {
...
+      res.status(404).send({ message: "please provide addressIds to delete the addresses" })
+    } else {
+      const userAllAddresses = await addressModel.findAll({ where: { user_id } })
```
Comment (amrit56789) [inline controllers/userController.js:138]: there is no need of findAll method

File: controllers/userController.js [hunk 4]
```diff
@@ -129,5 +129,24 @@ const userAddress = async (req, res) => {
...
+      const userAllAddressesIds = userAllAddresses.map((address) => address.id)
```
Comment (amrit56789) [inline controllers/userController.js]: don't use map function there is inbuilt method in sequelize use that

File: controllers/userController.js [hunk 5]
```diff
@@ -129,5 +129,24 @@ const userAddress = async (req, res) => {
...
+      const deletedAddresses = await addressModel.destroy({ where: { id: addressArray, user_id } })
```
Comment (amrit56789) [inline controllers/userController.js:140]: dont use addressArray use only user_id

File: controllers/userController.js [hunk 6]
```diff
@@ -129,5 +129,24 @@ const userAddress = async (req, res) => {
...
+      if (deletedAddresses > 0) {
```
Comment (amrit56789) [inline controllers/userController.js:141]: also remove this

File: controllers/userController.js [hunk 7]
```diff
@@ -129,5 +130,34 @@ const userAddress = async (req, res) => {
     res.status(500).send({ message: err })
   }
 }
-
-module.exports = { userRegister, login, getUserData, deleteUserData, limitUsersData, userAddress }
+const deleteUserAddress = async (req, res) => {
+  const { addressArray, user_id } = req.body
```
Comment (amrit56789) [inline controllers/userController.js:134]: why are you using addressArray there is no need you have to take only user_id

File: controllers/userController.js [hunk 8]
```diff
@@ -129,5 +130,34 @@ const userAddress = async (req, res) => {
...
+  try {
+    if (addressArray.length == 0) {
```
Comment (amrit56789) [inline controllers/userController.js:136]: remove this.

File: controllers/userController.js [hunk 9]
```diff
@@ -129,5 +130,34 @@ const userAddress = async (req, res) => {
...
+      res.status(404).send({ message: "please provide userIds to delete the addresses" })
+    } else {
+      const countAddress = await addressModel.count({
```
Comment (amrit56789) [inline controllers/userController.js:139]: There is no need to count the addresses.

File: controllers/userController.js [hunk 10]
```diff
@@ -129,5 +130,34 @@ const userAddress = async (req, res) => {
     res.status(500).send({ message: err })
   }
 }
-
-module.exports = { userRegister, login, getUserData, deleteUserData, limitUsersData, userAddress }
+const deleteUserAddress = async (req, res) => {
```
Comment (amrit56789) [inline controllers/userController.js]: Change the variable name you are deleting multiple user

File: controllers/userController.js [hunk 11]
```diff
@@ -129,5 +130,34 @@ const userAddress = async (req, res) => {
...
+        where: {
+          user_id: {
+            [Op.in]: addressArray
+          }
+        }
+      });
+      const deletedAddresses = await addressModel.destroy({ where: { user_id: addressArray } })
+      if (deletedAddresses > 0) {
+        let addressDeletedMessage;
+        if (countAddress != addressArray.length) {
```
Comment (amrit56789) [inline controllers/userController.js:149]: There is no need of this condition

File: controllers/userController.js [hunk 12]
```diff
@@ -129,5 +130,34 @@ const userAddress = async (req, res) => {
...
+const deleteUserAddress = async (req, res) => {
+  const { addressArray, user_id } = req.body
+  try {
+    if (addressArray.length == 0) {
+      res.status(404).send({ message: "please provide userIds to delete the addresses" })
+    } else {
+      const countAddress = await addressModel.count({
+        where: {
+          user_id: {
+            [Op.in]: addressArray
```
Comment (amrit56789) [inline controllers/userController.js]: check more OP methods

File: controllers/userController.js [hunk 13]
```diff
@@ -129,5 +129,24 @@ const userAddress = async (req, res) => {
...
 }
-
-module.exports = { userRegister, login, getUserData, deleteUserData, limitUsersData, userAddress }
+const deleteUserAddresses = async (req, res) => {
+  const { userAddresses, user_id } = req.body
+  try {
+    if (userAddresses.length === 0) {
+      res.status(404).send({ message: "please provide addressIds to delete the addresses" })
+    } else {
+      const findAllUserAddresses = await addressModel.findAll({ where: { user_id } })
```
Comment (yuvrajInno) [inline controllers/userController.js]: why do you need to get all user address? when api having address's ids in payload? @ashok-choudhary-0

File: controllers/userController.js [hunk 14]
```diff
@@ -129,5 +129,24 @@ const userAddress = async (req, res) => {
...
+      const userAllAddressesIds = findAllUserAddresses.map((obj) => obj.id)
```
Comment (yuvrajInno) [inline controllers/userController.js]: these id should be coming from payload @ashok-choudhary-0

**Review by yuvrajInno — CHANGES_REQUESTED** [review]

**Pull request merged**